v4/source/posts.yaml

```
@@ -216,6 +216,36 @@
         '403':
           $ref: '#/responses/Forbidden'
 
+  '/posts/{post_id}/set-unread':
```

```suggestion
  '/posts/{post_id}/set_unread':
```

Minor but should be underscores for consistency

---

Actually, this is marking the channel/post unread for a specific user right? I assume you're just using the logged in user but we try to supply the user ID as part of API requests anyways so that admins/integrations with the right permissions could mark the posts as unread

So then it would be something like 
```suggestion
  '/users/{user_id}/posts/{post_id}/set_unread':
```

That make sense?

---

I like the idea because of consistency.
One thing: do you want to implement controls for an admin to get to do that, or just theorising about possibilities?
I can't find any reason for an admin to do that, but you might have any.

---

We've gotten in the habit of including a user ID lately for new APIs in case someone comes up with a reason to do that later on

---

yup, I got that, I was just curious if there was a use case, to make sure and implement it on the server :)
anyways it's already modified in both places to take the user id

---

I think it's worth the extra effort to support admins specifying other user IDs. I could definitely see it used by an integration that works with unreads/notifications